### PR TITLE
#944 Substitution of MatcherAsssert.assertThat () by Assertion

### PR DESCRIPTION
**Pull request by marceloamadeu**
For #944:

- Substitution of some `MatcherAsssert.assertThat` by `Assertion` (`org.llorllale.cactoos.matchers`).

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#963](https://github.com/yegor256/takes/issues/963) is now in scope, role is `REV`

**Comment by marceloamadeu** [comment]
@paulodamaso please take a look at this

File: src/test/java/org/takes/facets/auth/PsAllTest.java [hunk 1]
```diff
@@ -27,27 +27,29 @@
...
      * Fails if PsAll with 0 Passes is created.
      * @throws Exception If fails
      */
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void thereShouldBeAtLeastOnePass() throws Exception {
-        MatcherAssert.assertThat(
-            new PsAll(
+        new Assertion<>(
+            "An error occurred while executing thereShouldBeAtLeastOnePass()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/auth/PsAllTest.java]: @marceloamadeu Please make this message better, it should show us why the assertion failed.

File: src/test/java/org/takes/facets/auth/PsAllTest.java [hunk 2]
```diff
@@ -59,10 +61,11 @@ public void thereShouldBeAtLeastOnePass() throws Exception {
      * Fails if index is less then 0.
      * @throws Exception If fails
      */
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void indexMustBeNonNegative() throws Exception {
-        MatcherAssert.assertThat(
-            new PsAll(
+        new Assertion<>(
+            "An error occurred while executing indexMustBeNonNegative()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/auth/PsAllTest.java]: @marceloamadeu Here too.

File: src/test/java/org/takes/facets/auth/PsAllTest.java [hunk 3]
```diff
@@ -74,10 +77,11 @@ public void indexMustBeNonNegative() throws Exception {
      * Fails if index is greater or equal to the number of Passes to enter.
      * @throws Exception If fails
      */
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void indexMustBeSmallEnough() throws Exception {
-        MatcherAssert.assertThat(
-            new PsAll(
+        new Assertion<>(
+            "An error occurred while executing indexMustBeSmallEnough()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/auth/PsAllTest.java]: @marceloamadeu Here again..

File: src/test/java/org/takes/facets/auth/PsAllTest.java [hunk 4]
```diff
@@ -94,8 +98,9 @@ public void indexMustBeSmallEnough() throws Exception {
      */
     @Test
     public void testOneSuccessfull() throws Exception {
-        MatcherAssert.assertThat(
-            new PsAll(
+        new Assertion<>(
+            "An error occurred while executing testOneSuccessfull()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/auth/PsAllTest.java]: @marceloamadeu Here too.

File: src/test/java/org/takes/facets/auth/PsAllTest.java [hunk 5]
```diff
@@ -109,8 +114,9 @@ public void testOneSuccessfull() throws Exception {
      */
     @Test
     public void testOneFail() throws Exception {
-        MatcherAssert.assertThat(
-            new PsAll(
+        new Assertion<>(
+            "An error occurred while executing testOneFail()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/auth/PsAllTest.java]: @marceloamadeu And here.

File: src/test/java/org/takes/facets/auth/PsAllTest.java [hunk 6]
```diff
@@ -129,8 +135,9 @@ public void testSuccessfullIdx() throws Exception {
             new Identity.Simple("urn:foo:test")
         );
         final RqFake request = new RqFake();
-        MatcherAssert.assertThat(
-            new PsAll(
+        new Assertion<>(
+            "An error occurred while executing testSuccessfullIdx()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/auth/PsAllTest.java]: @marceloamadeu Again here.

File: src/test/java/org/takes/facets/auth/PsAllTest.java [hunk 7]
```diff
@@ -149,8 +156,9 @@ public void testSuccessfullIdx() throws Exception {
      */
     @Test
     public void testFail() throws Exception {
-        MatcherAssert.assertThat(
-            new PsAll(
+        new Assertion<>(
+            "An error occurred while executing testFail()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/auth/PsAllTest.java]: @marceloamadeu Rewrite this too

File: src/test/java/org/takes/facets/auth/PsAllTest.java [hunk 8]
```diff
@@ -171,8 +179,9 @@ public void testFail() throws Exception {
     public void exits() throws Exception {
         final Response response = new RsEmpty();
         final PsFake exiting = new PsFake(true);
-        MatcherAssert.assertThat(
-            new PsAll(
+        new Assertion<>(
+            "An error occurred while executing exits()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/auth/PsAllTest.java]: @marceloamadeu And here too

File: src/test/java/org/takes/rs/RsPrettyXmlTest.java [hunk 9]
```diff
@@ -27,30 +27,32 @@
...
     /**
      * RsPrettyXML can format response with XML body.
      * @throws IOException If some problem inside
      */
     @Test
     public void formatsXmlBody() throws IOException {
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing formatsXmlBody()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsPrettyXmlTest.java]: @marceloamadeu See above

File: src/test/java/org/takes/rs/RsPrettyXmlTest.java [hunk 10]
```diff
@@ -60,15 +62,16 @@ public void formatsXmlBody() throws IOException {
     @Test
     // @checkstyle MethodNameCheck (1 line)
     public void formatsHtml5DoctypeBody() throws IOException {
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing formatsHtml5DoctypeBody()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsPrettyXmlTest.java]: @marceloamadeu See above

File: src/test/java/org/takes/rs/RsPrettyXmlTest.java [hunk 11]
```diff
@@ -80,8 +83,9 @@ public void formatsHtml5DoctypeBody() throws IOException {
     @Test
     // @checkstyle MethodNameCheck (1 line)
     public void formatsHtml5ForLegacyBrowsersDoctypeBody() throws IOException {
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsPrettyXmlTest.java]: @marceloamadeu See above

File: src/test/java/org/takes/rs/RsPrettyXmlTest.java [hunk 12]
```diff
@@ -92,7 +96,7 @@ public void formatsHtml5ForLegacyBrowsersDoctypeBody() throws IOException {
                     )
                 )
             ).printBody(),
-            Matchers.is(
+            new IsEqual<>(
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsPrettyXmlTest.java]: @marceloamadeu Use `JoinedText` here too

File: src/test/java/org/takes/rs/RsPrettyXmlTest.java [hunk 13]
```diff
@@ -119,8 +123,9 @@ public void formatsHtml4DoctypeBody() throws IOException {
         final String pid = "PUBLIC \"-//W3C//DTD HTML 4.01 Transitional//EN\" ";
         final String xhtml = "<html xmlns=\"http://www.w3.org/1999/xhtml\" "
             .concat("lang=\"en\">");
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing formatsHtml4DoctypeBody()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsPrettyXmlTest.java]: @marceloamadeu See above about rewriting those texts

File: src/test/java/org/takes/rs/RsPrettyXmlTest.java [hunk 14]
```diff
@@ -134,7 +139,7 @@ public void formatsHtml4DoctypeBody() throws IOException {
                     )
                 )
             ).printBody(),
-            Matchers.is(
+            new IsEqual<>(
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsPrettyXmlTest.java]: @marceloamadeu Use `JoinedText` here

File: src/test/java/org/takes/rs/RsPrettyXmlTest.java [hunk 15]
```diff
@@ -170,8 +175,9 @@ public void reportsCorrectContentLength() throws IOException {
                 "<test>\n   <a>test</a>\n</test>\n"
             )
         ).printBody(baos);
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing reportsCorrectContentLength()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsPrettyXmlTest.java]: @marceloamadeu Again, rewrite this text too

File: src/test/java/org/takes/rs/RsWithStatusTest.java [hunk 16]
```diff
@@ -44,14 +45,15 @@
      */
     @Test
     public void addsStatus() throws IOException {
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing addsStatus()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithStatusTest.java]: @marceloamadeu Rewrite this too

File: src/test/java/org/takes/rs/RsWithStatusTest.java [hunk 17]
```diff
@@ -44,14 +45,15 @@
...
+            () -> new RsPrint(
                 new RsWithStatus(
                     new RsWithHeader("Host", "example.com"),
                     HttpURLConnection.HTTP_NOT_FOUND
                 )
             ).print(),
-            Matchers.equalTo(
+            new IsEqual<>(
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithStatusTest.java]: @marceloamadeu Use `JoinedText` here

File: src/test/java/org/takes/rs/RsWithStatusTest.java [hunk 18]
```diff
@@ -75,12 +77,9 @@ public void addsStatusMultipleTimes() throws IOException {
...
             HttpURLConnection.HTTP_SEE_OTHER
         );
-        MatcherAssert.assertThat(
-            response.head(),
-            Matchers.<String>iterableWithSize(1)
-        );
-        MatcherAssert.assertThat(
-            response.head(),
+        new Assertion<>(
+            "An error occurred while executing addsStatusMultipleTimes()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithStatusTest.java]: @marceloamadeu as above, about the text

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 19]
```diff
@@ -85,8 +85,9 @@
     @Test
     public void replaceTypeToResponse() throws Exception {
         final String type = RsWithTypeTest.TYPE_TEXT;
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing replaceTypeToResponse()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu Lets rewrite this

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 20]
```diff
@@ -111,8 +112,9 @@ public void replaceTypeToResponse() throws Exception {
     @Test
     public void doesNotReplaceResponseCode() throws Exception {
         final String body = "Error!";
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing doesNotReplaceResponseCode()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu Rewrite here too

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 21]
```diff
@@ -140,8 +142,9 @@ public void doesNotReplaceResponseCode() throws Exception {
      */
     @Test
     public void replacesTypeWithHtml() throws Exception {
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing replacesTypeWithHtml()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu Here too

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 22]
```diff
@@ -159,8 +162,9 @@ public void replacesTypeWithHtml() throws Exception {
                 )
             )
         );
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing replacesTypeWithHtml() - ISO",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu And here

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 23]
```diff
@@ -188,8 +192,9 @@ public void replacesTypeWithHtml() throws Exception {
      */
     @Test
     public void replacesTypeWithJson() throws Exception {
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing replacesTypeWithJson()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu And here

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 24]
```diff
@@ -207,8 +212,9 @@ public void replacesTypeWithJson() throws Exception {
                 )
             )
         );
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing replacesTypeWithJson() - ISO",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu Again, here too

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 25]
```diff
@@ -236,8 +242,9 @@ public void replacesTypeWithJson() throws Exception {
      */
     @Test
     public void replacesTypeWithXml() throws Exception {
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing replacesTypeWithXml()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu Rewrite this message here too

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 26]
```diff
@@ -254,8 +261,9 @@ public void replacesTypeWithXml() throws Exception {
                 )
             )
         );
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing replacesTypeWithXml() - ISO",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu Rewrite this message here too

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 27]
```diff
@@ -283,8 +291,9 @@ public void replacesTypeWithXml() throws Exception {
      */
     @Test
     public void replacesTypeWithText() throws Exception {
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing replacesTypeWithText()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu Rewrite this message here too

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 28]
```diff
@@ -301,8 +310,9 @@ public void replacesTypeWithText() throws Exception {
                 )
             )
         );
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing replacesTypeWithText() - ISO",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu Rewrite this message here too

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 29]
```diff
@@ -330,8 +340,9 @@ public void replacesTypeWithText() throws Exception {
      */
     @Test
     public void addsContentType() throws Exception {
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing addsContentType()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu Rewrite this message here too

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 30]
```diff
@@ -356,8 +367,9 @@ public void addsContentType() throws Exception {
      */
     @Test
     public void addsCharsetToContentType() throws Exception {
-        MatcherAssert.assertThat(
-            new RsPrint(
+        new Assertion<>(
+            "An error occurred while executing addsCharsetToContentType()",
```
Comment (paulodamaso) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @marceloamadeu Rewrite this message here too

**Comment by 0crat** [comment]
The job [#963](https://github.com/yegor256/takes/issues/963) assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C3FFK3YAY/7c88d57b-ae65-495d-b3bd-3f012ad66d4d); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by paulodamaso** [comment]
@fabriciofx ping

**Comment by 0crat** [comment]
Job [#963](https://github.com/yegor256/takes/issues/963) is not in the agenda of @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), can't retrieve data and time of add

**Comment by 0crat** [comment]
Job [#963](https://github.com/yegor256/takes/issues/963) is not in the agenda of @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), can't inspect

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (331 days), architects (@paulodamaso) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Pull request [#963](https://github.com/yegor256/takes/issues/963) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Pull request closed without merging**